### PR TITLE
[14.0][ADD] stock_mts_mto_mrp_rule

### PR DESCRIPTION
**Pull request by geomer198**
Supersedes #1301

This PR add the functionality to use mts+mto in procurements created from MRP module.

### CODE REVIEW

File: stock_mts_mto_mrp_rule/tests/test_stock_move.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from odoo.addons.stock.tests.common import TestStockCommon
+
+
+class TestStockMove(TestStockCommon):
+    pass
```
Comment (rousseldenis) [inline stock_mts_mto_mrp_rule/tests/test_stock_move.py]: @geomer198 Do you plan to update this ? :smiley:

Comment (geomer198) [reply stock_mts_mto_mrp_rule/tests/test_stock_move.py]: @rousseldenis Of Course!

File: stock_mts_mto_mrp_rule/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "Stock MTS+MTO MRP Rule",
+    "summary": "Stock MTS+MTO MRP Rule",
+    "version": "14.0.1.0.0",
+    "development_status": "Mature",
```
Comment (rousseldenis) [inline stock_mts_mto_mrp_rule/__manifest__.py]: IMHO, it's quite soon to set it like that.
Remove the entry as default one is Beta.

File: stock_mts_mto_mrp_rule/demo/demo.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo noupdate="1">
+    <record id="stock_location_route_dummy_route" model="stock.location.route">
```
Comment (rousseldenis) [inline stock_mts_mto_mrp_rule/demo/demo.xml]: It's better to create those 'dummy' data in setupClass of tests if the aim is to use them there only.

In demo, should appear only real 'demo' data (with significant meaning - something you want to show to user).

File: stock_mts_mto_mrp_rule/models/stock_move.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            domain = [
+                ("location_src_id", "=", move.location_id.id),
+                ("location_id", "=", move.location_dest_id.id),
+                ("action", "!=", "push"),
+            ]
+            rules = self.env["procurement.group"]._search_rule(
+                False, product_id, move.warehouse_id, domain
+            )
+            if not rules or rules and rules.action != "split_procurement":
+                return super(StockMove, self)._adjust_procure_method()
```
Comment (rousseldenis) [inline stock_mts_mto_mrp_rule/models/stock_move.py]: This is not correct as self != move

File: stock_mts_mto_mrp_rule/models/stock_move.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            if float_is_zero(needed_qty, precision_digits=precision):
+                move.procure_method = "make_to_stock"
+            elif (
+                float_compare(needed_qty, move.product_qty, precision_digits=precision)
+                == 0.0
+            ):
+                move.procure_method = "make_to_order"
+            else:
+                mts_qty = move.product_uom_qty - needed_qty
+                move.copy(
```
Comment (rousseldenis) [inline stock_mts_mto_mrp_rule/models/stock_move.py]: You do nothing with new copied move ? Is this really a copy or do you want to split quantities (as there is a split method) ?

File: stock_mts_mto_mrp_rule/tests/test_stock_move.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
+from odoo.tests import TransactionCase
+
+
+class TestStockMove(TransactionCase):
+    def setUp(self):
```
Comment (rousseldenis) [inline stock_mts_mto_mrp_rule/tests/test_stock_move.py]: Prefer using setupClass as it would be instantiated just once between all tests.

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

@geomer198 I would have added also readme fragments  with at least a description of what this module does functionally. Thanks

File: stock_mts_mto_mrp_rule/__manifest__.py (hunk 2 above)
Comment (geomer198) [reply stock_mts_mto_mrp_rule/__manifest__.py]: Resolved

File: stock_mts_mto_mrp_rule/demo/demo.xml (hunk 3 above)
Comment (geomer198) [reply stock_mts_mto_mrp_rule/demo/demo.xml]: Resolved. Moved demo data to TestCommon

File: stock_mts_mto_mrp_rule/models/stock_move.py (hunk 4 above)
Comment (geomer198) [reply stock_mts_mto_mrp_rule/models/stock_move.py]: Resolved

File: stock_mts_mto_mrp_rule/models/stock_move.py (hunk 5 above)
Comment (geomer198) [reply stock_mts_mto_mrp_rule/models/stock_move.py]: Resolved. I'm applied '_split' method from stock.move.

File: stock_mts_mto_mrp_rule/tests/test_stock_move.py (hunk 6 above)
Comment (geomer198) [reply stock_mts_mto_mrp_rule/tests/test_stock_move.py]: Resolved

**Comment by francesco-ooops** [comment]
> @geomer198 I would have added also readme fragments with at least a description of what this module does functionally. Thanks

@geomer198 could you add this too?

**Comment by geomer198** [comment]
> Code review.
> 
> @geomer198 I would have added also readme fragments with at least a description of what this module does functionally. Thanks

Hello @rousseldenis ! Can you check changes?

**Comment by francesco-ooops** [comment]
@alan196 could you review? thanks!

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by elvise — APPROVED** [review]
LGTM

**Comment by francesco-ooops** [comment]
@OCA/logistics-maintainers can we merge? Thanks!

**Pull request merged**